moveit_ros/planning_interface/moveit_cpp/include/moveit/moveit_cpp/planning_component.h

```diff
@@ -108,10 +108,21 @@ class PlanningComponent
   {
     std::string planner_id;
     std::string planning_pipeline;
-    size_t planning_attempts;
+    int planning_attempts;
     double planning_time;
     double max_velocity_scaling_factor;
     double max_acceleration_scaling_factor;
+
+    void load(const ros::NodeHandle& nh)
+    {
+      std::string ns = "plan_request_params/";
+      nh.param(ns + "planner_id", planner_id, std::string(""));
+      nh.param(ns + "planning_pipeline", planning_pipeline, std::string(""));
+      nh.param(ns + "planning_time", planning_attempts, 1);
+      nh.param(ns + "planning_attempts", planning_time, 5.0);
+      nh.param(ns + "max_velocity_scaling_factor", max_velocity_scaling_factor, 1.0);
+      nh.param(ns + "max_acceleration_scaling_factor", max_acceleration_scaling_factor, 1.0);
+    }
```

Comment:
What about error checking in this function? The `param` method will return false on failure but there is no checking for that here. The default value is then utilized but wouldn't it be better to inform user that the parameter values they set are not being used? For example, would it be possible to typo the `max_velocity_scaling_factor`, default to 1.0 and have the arm move really fast?

Reply:
I don't think error checking is required because of the default values. You could say that overriding these values is the optional behavior, so I don't think we need extensive user infos here. Sanity checks and warnings are issued by the planning pipeline in case something is wrong. In general, moveit_cpp is currently not meant to provide a fool-proof user interface, but I would agree that we could add more debugging output about configuration and checks.